CI/TST: stats: `TestMultivariateNormal` failing in py3.14-dev job

Example log https://github.com/scipy/scipy/actions/runs/24441141953/job/71406241984

```
FAILED scipy/stats/tests/test_multivariate.py::TestMultivariateNormal::test_fit_correctness[5] - AssertionError: 
Not equal to tolerance rtol=1e-15, atol=0

Mismatched elements: 1 / 25 (4%)
Mismatch at index:
 [4, 2]: 0.00634042985464201 (ACTUAL), 0.0063404298546420185 (DESIRED)
Max absolute difference among violations: 8.67361738e-18
Max relative difference among violations: 1.3679857e-15
 ACTUAL: array([[ 0.090903, -0.003803,  0.007249,  0.012642,  0.017278],
       [-0.003803,  0.087393, -0.003652, -0.005125,  0.002108],
       [ 0.007249, -0.003652,  0.097427, -0.007884,  0.00634 ],...
 DESIRED: array([[ 0.090903, -0.003803,  0.007249,  0.012642,  0.017278],
       [-0.003803,  0.087393, -0.003652, -0.005125,  0.002108],
       [ 0.007249, -0.003652,  0.097427, -0.007884,  0.00634 ],...
```

This started appearing on `main` at some point in the past 2 days[^1]:

<img width="768" height="430" alt="Image" src="https://github.com/user-attachments/assets/43e23a65-8feb-4bcc-8c1b-b1adf59d6941" />

and the situation is pretty dire on PRs:

<img width="760" height="435" alt="Image" src="https://github.com/user-attachments/assets/d6a82767-6a25-478f-befc-889aefac62f4" />

[^1]: https://actions-search.scicftest.com/actions/scipy/scipy/search/?is_trivial__false=1&name__contains=3.14-dev%2C+False&branch__eq=main

--- a/scipy/stats/tests/test_multivariate.py
+++ b/scipy/stats/tests/test_multivariate.py
@@ -1158,7 +1158,7 @@ class TestMultivariateNormal:
         mean_est, cov_est = multivariate_normal.fit(x)
         mean_ref, cov_ref = np.mean(x, axis=0), np.cov(x.T, ddof=0)
         assert_allclose(mean_est, mean_ref, atol=1e-15)
-        assert_allclose(cov_est, cov_ref, rtol=1e-15)
+        assert_allclose(cov_est, cov_ref, rtol=5e-15)
 
     def test_fit_both_parameters_fixed(self):
         data = np.full((2, 1), 3)

